src/views/signUpBuyer.jsx

```diff
@@ -0,0 +1,136 @@
+          method: 'POST',
+          mode: 'cors',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            firstname: e.target[0].value,
+            lastname: e.target[1].value,
+            email: e.target[2].value,
+            phone: e.target[3].value,
+            password: e.target[4].value,
+          }),
+        })
+          .then((res) => res.json())
+          .then((res) => {
+            console.log(res);
+            if (res.status[0] == 's') {
+              toast.success(res.message, { theme: 'colored' });
+              setTimeout(() => {
+                navigate('/login');
+              }, 2000);
+            } else {
+              toast.error(res.message, { theme: 'colored' });
+            }
+          });
+      } else {
+        e.target[5].focus();
+        toast.error('confirm the correct password', { theme: 'colored' });
+      }
+    }
+    setbtn({
+      display: 'hidden',
+      text: `Sign Up`,
+    });
+  };
+
+  return (
+    <div className="bg-[#D9D9D9]  min-h-[100vh] pt-20 ">
+      <ToastContainer />
+      <div className="lg:w-full  w-[70em] h-[70em]  bg-[#011B32] fixed -right-[35vw] rounded-[50%]  -top-[20vh] overflow-hidden "></div>
+      <div className=" flex flex-row-reverse justify-center mx-auto ">
+        <img
+          src={CART}
+          alt="cart image"
+          className="lg:hidden grid backdrop-blur-3xl bg-[#D9D9D9] w-[28em] px-6 py-14 rounded-sm rounded-l-none shadow-2xl"
+        />
+        <div className=" lg:w-[95vw] min-w-[270px] max-w-[30em] w-[40vw] lg:px-4 px-6 h-fit py-10 pt-5 bg-[white] backdrop-blur-md rounded-md lg:rounded-r rounded-r-none shadow-2xl flex flex-col ">
+          <h1 className="h-fit text-2xl font-serif font-medium text-[#011B32]">
+            Excellent shoppify experience
+          </h1>
+          <form onSubmit={SignUp}>
+            <p className="font-serif ">
+              please fill the fields below to have an account
+            </p>
+            {allFields.map(({ placeh, label, type }) => {
+              return (
+                <div
+                  key={placeh}
+                  className="relative  w-full min-w-[200px] mt-1 mb-6 h-[2em]"
+                >
+                  <label className=" text-[#2b2b2b] text-sm font-serif">
+                    {label}
+                  </label>
+                  <input
+                    className="invalid:focus:border-red-400 valid:focus:border-green-400 h-full w-full rounded-[5px] border border-blue-gray-200 px-3 py-4 font-sans text-md font-extralight text-blue-gray-400 outline-none outline-[.2px] transition-all  placeholder-shown:border-t-blue-gray-200 focus:border-1 focus:border-blue-500 focus:outline-0  "
+                    placeholder={placeh}
+                    type={type}
+                    required
+                  />
+                </div>
+              );
+            })}
```

Comment:
Could you please define the Custom Component for this map. It'll help to have reusable components.